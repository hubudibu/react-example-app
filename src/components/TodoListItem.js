import React from 'react';

const TodoListItem = (props) => <li>{props.todo}</li>;

export default TodoListItem;
