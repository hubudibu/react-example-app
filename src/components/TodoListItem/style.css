.TodoListItem {
  list-style: none;
  margin-bottom: .5em;
}
