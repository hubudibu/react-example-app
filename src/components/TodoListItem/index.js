import React from 'react';
import './style.css';

const TodoListItem = (props) => <li className="TodoListItem">{props.todo && props.todo.text}</li>;

export default TodoListItem;
