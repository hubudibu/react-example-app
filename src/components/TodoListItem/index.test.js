import React from 'react';
import renderer from 'react-test-renderer';
import TodoListItem from './';

describe('Todo List Item', () => {
  it('renders empty item as in snapshot', () => {
    const wrapper = renderer.create(<TodoListItem />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders todo item as in snapshot', () => {
    const todo = {
      text: 'Sample todo'
    };
    const wrapper = renderer.create(<TodoListItem todo={todo} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
})
