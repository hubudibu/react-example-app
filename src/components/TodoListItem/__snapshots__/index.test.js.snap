exports[`Todo List Item renders empty item as in snapshot 1`] = `
<li
  className="TodoListItem" />
`;

exports[`Todo List Item renders todo item as in snapshot 1`] = `
<li
  className="TodoListItem">
  Sample todo
</li>
`;
